ibis/backends/dask/udf.py

```diff
@@ -0,0 +1,161 @@
+
+    @execute_node.register(
+        ops.ElementWiseVectorizedUDF,
+        *(itertools.repeat(ddgb.SeriesGroupBy, nargs)),
+    )
+    def execute_udf_node_groupby(op, *args, **kwargs):
+        func = op.func
+
+        # all grouping keys must be identical
+        assert_identical_grouping_keys(*args)
+
+        # we're performing a scalar operation on grouped column, so
+        # perform the operation directly on the underlying Series
+        # and regroup after it's finished
+        args_objs = [make_selected_obj(arg) for arg in args]
+        groupings = args[0].index
+        return dd.map_partitions(func, *args_objs).groupby(groupings)
+
+    # Define an execution rule for a simple elementwise Series
+    # function
+    @execute_node.register(
+        ops.ElementWiseVectorizedUDF, *(itertools.repeat(dd.Series, nargs))
+    )
+    def execute_udf_node(op, *args, **kwargs):
+        # We have rewritten op.func to be a closure enclosing
+        # the kwargs, and therefore, we do not need to pass
+        # kwargs here. This is true for all udf execution in this
+        # file.
+        # See ibis.udf.vectorized.UserDefinedFunction
+        if isinstance(op._output_type, dt.Struct):
+            # dask does not know how to glue together Tuple[*pd.Series].
+            # We create a wrapper that performs the "gluing" in each partition
+            def wrapper(func, *cols):
+                raw = func(*cols)
+                return pandas.Series(zip(*raw))
+
+            df = dd.map_partitions(wrapper, op.func, *args, meta="object")
+            df.index = args[0].index
+            return df
+        else:
+            df = dd.map_partitions(
+                op.func, *args, meta=op._output_type.to_dask()
+            )
+
+            return df
+
+    @execute_node.register(
+        ops.ElementWiseVectorizedUDF, *(itertools.repeat(object, nargs))
+    )
+    def execute_udf_node_non_dask(op, *args, **kwargs):
+        return op.func(*args)
+
+    return scope
+
+
+@pre_execute.register(ops.AnalyticVectorizedUDF)
+@pre_execute.register(ops.AnalyticVectorizedUDF, ibis.client.Client)
+@pre_execute.register(ops.ReductionVectorizedUDF)
+@pre_execute.register(ops.ReductionVectorizedUDF, ibis.client.Client)
+def pre_execute_analytic_and_reduction_udf(op, *clients, scope=None, **kwargs):
+    input_type = op.input_type
+    nargs = len(input_type)
+
+    # An execution rule to handle analytic and reduction UDFs over
+    # 1) an ungrouped window,
+    # 2) an ungrouped Aggregate node, or
+    # 3) an ungrouped custom aggregation context
+    @execute_node.register(type(op), *(itertools.repeat(dd.Series, nargs)))
+    def execute_udaf_node_no_groupby(op, *args, aggcontext, **kwargs):
+
+        meta = pandas.Series([], dtype=op._output_type.to_dask())
+
+        @dask.delayed
+        def lazy_agg(*series):
+            return aggcontext.agg(series[0], op.func, *series[1:])
+
+        lazy_result = lazy_agg(*args)
```

Comment:
Discussed offline. Let's put some comments here to explain what is happening:

dask.delayed will turn dd.Series into pd.Series (by partition) and pass down to actual lazy_agg function.

Reply:
Per discussion this doesn't do map partitions, it just collect the entire dd.Series to a single worker as pd.Series